tools/src/h5ls/h5ls.h

```diff
@@ -70,6 +70,9 @@
  * \li <strong>--page-buffer-size=N</strong> Set the page buffer cache size, N=non-negative integers
  * \li <strong>--vfd=DRIVER</strong>   Use the specified virtual file driver
  * \li <strong>--hexdump</strong>      Show raw data in hexadecimal format
+ * \li <strong>--endpoint-url=P</strong> Supply S3 endpoint url information to "ros3" vfd.
+ *                   P is the AWS service endpoint.
+ *                   Has no effect if vfd flag not set to "ros3".
```

Comment:
I'm very much against continuing to add options to the tools that are only relevant to one particular VFD. If there's a good use case for it, I think the whole "endpoint URL" feature should be split out into a separate PR and discussed there. The idea of updating the VFD interface to account for VFD-specific configuration was brought up somewhat recently and I think this is a good fit for that.